pkg/scheduler/apis/config/types_pluginargs.go

```diff
@@ -52,6 +52,10 @@ type InterPodAffinityArgs struct {
 	// HardPodAffinityWeight is the scoring weight for existing pods with a
 	// matching hard affinity to the incoming pod.
 	HardPodAffinityWeight int32
+
+	// IgnorePreferredTermsOfExistingPods configures the scheduler to ignore existing pods' preferred affinity
+	// rules when scoring candidate nodes, unless the incoming pod has inter-pod affinities.
+	IgnorePreferredTermsOfExistingPods bool
```

Comment:
the "unless incoming pod..." aspect is not captured in the name... would we eventually want to also allow unconditionally ignoring preferred terms of existing pods? if so, it would be worth thinking about how we would name this field and that eventual field in ways that wouldn't be confusing

Reply:
Based on my understanding, unconditionally ignoring preferred terms of existing pods is not something we'll want to do in the future. However, @ahg-g has more knowledge about future plans in this area so I will defer to his recommendation here.

Reply:
Right, I don't think we will want to do that